pkg/repository/manifest.go

```diff
@@ -243,6 +340,53 @@ func (manifest *Timestamp) Filename() string {
 	return "timestamp.json"
 }
 
+// SetRoles populates role list in the root manifest
+func (manifest *Root) SetRoles(manifestList []ValidManifest) {
+	if manifest.Roles == nil {
+		manifest.Roles = make(map[string]*Role)
+	}
+	for _, m := range manifestList {
+		manifest.Roles[m.Filename()] = &Role{
```

Comment:
I think it is better to iterate over `types` rather than passing in a manifest list (skip the component type because `filename == ""`). This is because in TUF, roles are not exactly the same as manifests. Alos, once there are component manifests around, there won't be a manifest list.